cpp/proto/serializer.h

```diff
@@ -307,64 +255,37 @@ class Serializer {
-
-  static SerializeResult SerializeV1SignedCertEntryWithType(
-      const std::string& leaf_certificate, std::string* result);
-
-  static SerializeResult SerializeV1SignedPrecertEntryWithType(
-      const std::string& issuer_key_hash, const std::string& tbs_certificate,
-      std::string* result);
-  static SerializeResult SerializeV2SignedPrecertEntryWithType(
-      const std::string& issuer_key_hash, const std::string& tbs_certificate,
-      std::string* result);
-
-  static SerializeResult SerializeV1SignedXJsonEntryWithType(
-      const std::string& json, std::string* result);
-
  private:
   // This class is mostly a namespace for static methods.
   // TODO(pphaneuf): Make this into normal functions in a namespace.
   Serializer() = delete;
 
-  // Serialize (with length prefix).
-  // FIXME(ekasper): for simplicity these reject if the list has empty
-  // elements (all our use cases are like this) but they should take in
-  // an arbitrary min bound instead.
-  static SerializeResult SerializeList(const repeated_string& in,
-                                       size_t max_elem_length,
-                                       size_t max_total_length,
-                                       std::string* result);
-  static SerializeResult CheckKeyHashFormat(const std::string& key_hash);
-
-  static SerializeResult CheckCertificateFormat(const std::string& cert);
-
-  static SerializeResult CheckSthExtensionsFormat(
-      const google::protobuf::RepeatedPtrField<ct::SthExtension>& extension);
-
-  static SerializeResult CheckChainFormat(const repeated_string& chain);
-
-  static SerializeResult CheckX509ChainEntryFormatV1(
-      const ct::X509ChainEntry& entry);
-  static SerializeResult CheckX509ChainEntryFormatV2(
-      const ct::X509ChainEntry& entry);
-
-  static SerializeResult CheckPrecertChainEntryFormatV1(
-      const ct::PrecertChainEntry& entry);
-  static SerializeResult CheckPrecertChainEntryFormatV2(
-      const ct::PrecertChainEntry& entry);
+  static std::function<std::string(const ct::LogEntry&)> leaf_data_;
+  static std::function<SerializeResult(
+      const ct::SignedCertificateTimestamp& sct, const ct::LogEntry& entry,
+      std::string* result)>
+      serialize_sct_sig_input_;
+  static std::function<SerializeResult(
+      const ct::SignedCertificateTimestamp& sct, const ct::LogEntry& entry,
+      std::string* result)>
+      serialize_sct_merkle_leaf_;
+  static std::function<SerializeResult(uint64_t timestamp, int64_t tree_size,
+                                       const std::string& root_hash,
+                                       std::string* result)>
+      serialize_sth_sig_input_v1_;
+  static std::function<SerializeResult(
+      uint64_t timestamp, int64_t tree_size, const std::string& root_hash,
+      const repeated_sth_extension& sth_extension, const std::string& log_id,
+      std::string* result)>
+      serialize_sth_sig_input_v2_;
```

Comment:
These are a fair bit of innards showing, you could remove those from here, I think?